src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+
+        result.push({ ...stateCopy });
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+
+        result.push({ ...stateCopy });
+        break;
+
+      case 'clear':
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
+
+        result.push({ ...stateCopy });
+        break;
+
+      default:
+        break;
```

Comment:
here you need to show the error using `throw`

Reply:
use 'throw new Error', read more here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw